### ISSUE TITLE
Black Powder is Impossible to Obtain.

### ISSUE THREAD

**Issue by necromanceranne** [tgstation/tgstation#47281]
For some reason, black powder has become black CRAYON powder. While it appears to be named Black Powder, it has the functionality of black crayon powder.

This means no black powder bombs and possibly cherry bombs outright don't work.

**Comment by necromanceranne** [comment]
Okay lemme close this and properly put up an issue. I had been spawning it but was told by a few people they weren't able to mix it recently. Potentially they were confusing it for the time during the fermichem testmerge.

**Comment by necromanceranne** [comment]
Fair

**Comment by Ty-the-Smonk** [comment]
This can probably be fixed by changing the crayon powders name

**Comment by ATH1909** [comment]
Oh, I think this one was caused by me in this PR: https://github.com/tgstation/tgstation/pull/46338.

Hold on, I'll rename "black powder" to "dark powder" in a new PR.

**Comment by ATH1909** [comment]
Bam, done: https://github.com/tgstation/tgstation/pull/47286.

**Comment by ATH1909** [comment]
@necromanceranne I thought that your idea for how to fix the problem was a good one as well, so I made an alternative PR that implements your idea instead of mine: https://github.com/tgstation/tgstation/pull/47288

**Comment by 81Denton** [comment]
```
/proc/build_name2reagent()
	. = list()
	for (var/t in subtypesof(/datum/reagent))
		var/datum/reagent/R = t
		if (length(initial(R.name)))
			.[ckey(initial(R.name))] = t
```

it hurts

**Comment by 81Denton** [comment]
tldr it checks the reagent's name instead of the typepath because???
